lib/node_modules/@stdlib/math/generics/random/sample/lib/factory.js

```diff
@@ -20,35 +25,119 @@ var slice = Array.prototype.slice;
 /**
 * Returns a function to sample elements from an array-like object.
 *
-* @param {PositiveInteger} seed - integer-valued seed
+* @param {ArrayLike} [pool] - array-like object to sample from
+* @param {Options} [options] - function options
+* @param {PositiveInteger} [options.seed] - integer-valued seed
+* @param {NonNegativeInteger} [options.size] - sample size
+* @param {boolean} [options.replace=true] - boolean indicating whether to sample with replacement
 * @returns {Function} function to sample elements from an array-like object
 *
 * @example
-* var sample = factory( 323 );
+* var sample = factory({
+*     'seed': 232
+* });
+* var out = sample( 'abcdefg' );
+* // [ 'g', 'd', 'g', 'f', 'c', 'e', 'f' ]
+*
+* @example
+* var sample = factory( [1,2,3,4,5,6], {
+*     'seed': 232
+*     'size': 2
+* });
+* var out = sample();
+* // returns [ 6, 4 ]
+*
+* out = sample()
+* // returns [ 6, 5 ]
+*
+* @example
+* var sample = factory( [1,2,3,4,5,6], {
+*     'seed': 474
+*     'size': 3
+*     'replace': false
+* });
+* var out = sample();
+* // returns [ 4, 3, 6 ]
+*
+* out = sample()
+* // returns [ 1, 5, 2 ]
+*
+* out = sample()
+* // throws <Error>
+*
+* @example
+* var sample = factory( [0,1], {
+*     'size': 2
+* });
+*
+* var out = sample();
+* // e.g., returns [ 1, 1 ]
+*
+* out = sample({
+*     size: 10
+* });
+* // e.g, returns [ 0, 1, 1, 1, 0, 1, 0, 0, 1, 1 ]
+*
+* @example
+* var sample = factory( [0,1], {
+*     'size': 2
+* });
+*
+* var out = sample();
+* // e.g., returns [ 1, 1 ]
+*
+* out = sample({
+*     'replace': false
+*     'size': 1
+* });
+* // returns [ 0 ] or [ 1 ]
+*
+* out = sample();
+* // throws <RangeError>
```

Comment:
My main thing is `replace=true` then `replace=false` does more than one thing:

1) Samples without replacement.
2) Mutates state.

The option is only explicit about the former, not the latter. Hence, the surprise. 

ATM we have no data to prefer one approach over the other. And we are in uncharted territory here as this implementation is unique in comparison to other platforms. I am fine supporting the current behavior, but would prefer if the behavior was explicitly specified.